src/org/traccar/smpp/SmppClient.java

```diff
@@ -184,31 +193,36 @@ private void destroySession() {
         }
     }
 
-    public synchronized void sendMessageSync(String destAddress, String message) throws RecoverablePduException,
-            UnrecoverablePduException, SmppTimeoutException, SmppChannelException, InterruptedException {
+    public synchronized void sendMessageSync(String destAddress, String message, boolean command)
+            throws RecoverablePduException, UnrecoverablePduException, SmppTimeoutException, SmppChannelException,
+            InterruptedException, IllegalStateException {
         if (getSession() != null && getSession().isBound()) {
-            byte[] textBytes = CharsetUtil.encode(message, charsetName);
-
             SubmitSm submit = new SubmitSm();
+            byte[] textBytes;
+            textBytes = CharsetUtil.encode(message, command ? commandsCharsetName : notificationsCharsetName);
+            submit.setDataCoding(command ? commandsDataCoding : notificationsDataCoding);
+            submit.setShortMessage(textBytes);
             submit.setSourceAddress(new Address(sourceTon, sourceNpi, sourceAddress));
             submit.setDestAddress(new Address(destTon, destNpi, destAddress));
-            submit.setDataCoding(dataCoding);
-            submit.setShortMessage(textBytes);
             SubmitSmResp submitResponce = getSession().submit(submit, submitTimeout);
-            Log.debug("SMS submited, msg_id: " + submitResponce.getMessageId());
+            if (submitResponce.getCommandStatus() == SmppConstants.STATUS_OK) {
+                Log.debug("SMS submited, msg_id: " + submitResponce.getMessageId());
```

Comment:
There is a typo here. Should be "submitted". Also, I think `msg_id` should be removed or converted to something more readable, like "id" or "message id".